src/Illuminate/Cache/RedisLock.php

```diff
@@ -80,4 +117,104 @@ public function getConnectionName()
     {
         return $this->redis->getName();
     }
+
+    /**
+     * Determines whether compression is enabled for the phpredis connection.
+     *
+     * @return bool
+     */
+    private function isPhpRedisCompressionEnabled()
+    {
+        $compressionOpt = constant('Redis::OPT_COMPRESSION');
+        $compressionNone = constant('Redis::COMPRESSION_NONE');
+
+        if ($this->redis->client()->getOption($compressionOpt) === $compressionNone) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Determines whether lzf compression is enabled for the phpredis connection.
+     *
+     * @return bool
+     */
+    private function isPhpRedisLzfEnabled()
+    {
+        $lzfConst = 'Redis::COMPRESSION_LZF';
+        if (! defined($lzfConst)) {
+            return false;
+        }
+
+        $lzfCompression = constant($lzfConst);
+        $compressionOpt = constant('Redis::OPT_COMPRESSION');
+
+        if ($this->redis->client()->getOption($compressionOpt) === $lzfCompression) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Determines whether zstd compression is enabled for the phpredis connection.
+     *
+     * @return bool
+     */
+    private function isPhpRedisZstdEnabled()
+    {
+        $zstdConst = 'Redis::COMPRESSION_ZSTD';
+        if (! defined($zstdConst)) {
+            return false;
+        }
+
+        $zstdCompression = constant($zstdConst);
+        $compressionOpt = constant('Redis::OPT_COMPRESSION');
+
+        if ($this->redis->client()->getOption($compressionOpt) === $zstdCompression) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Determines whether lz4 compression is enabled for the phpredis connection.
+     *
+     * @return bool
+     */
+    private function isPhpRedisLz4Enabled()
+    {
+        $lz4Const = 'Redis::COMPRESSION_LZ4';
+        if (! defined($lz4Const)) {
+            return false;
+        }
+
+        $lz4Compression = constant($lz4Const);
+        $compressionOpt = constant('Redis::OPT_COMPRESSION');
+
+        if ($this->redis->client()->getOption($compressionOpt) === $lz4Compression) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Determines whether the client used to connect to redis is one from the
+     * phpredis extension.
+     *
+     * @return bool
+     */
+    private function isPhpRedis()
+    {
+        $client = $this->redis->client();
+        $clientClass = get_class($client);
+        if ($clientClass === 'Redis' || $clientClass === 'RedisCluster') {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
Can the calls to `constant()` be avoided since you don't need to worry about classes `\Redis` and `\RedisCluster` not existing? Updating `isPhpRedis()` to use `instanceof`:

```php
use Redis;
use RedisCluster;

// ...

protected function isPhpRedis()
{
    $client = $this->redis->client();

    return $client instanceof Redis || $client instanceof RedisCluster;
}
```

This short-circuits the other `is*()` methods from being called. The `instanceof` keyword can accept any symbol, even for classes that don't exist. e.g., `$client instanceof ThisClassDoesNotExist` won't throw an exception - it just returns `false`.

Then the other predicate methods can be simplified to a single expression. e.g.,

```php
protected function isPhpRedisLz4Enabled()
{
    return defined('Redis::COMPRESSION_LZ4') &&
        $this->redis->client()->getOption(Redis::OPT_COMPRESSION) === Redis::COMPRESSION_LZ4;
}
```

Reply:
Sure, I was just not sure how static analyzers will react if classes/constants do not exist as they are added bases on compile config on phpredis